File: Data Structures & Algorithms/is-anagram/submission-4.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        d1, d2 = {}, {}

        for c in s:
            d1[c] = d1.get(c, 0) + 1
        
        for c in t:
            d2[c] = d2.get(c, 0) + 1
        return d1 == d2
        

File: Data Structures & Algorithms/string-encode-and-decode/submission-4.java
class Solution {

    public String encode(List<String> strs) {
        StringBuilder res = new StringBuilder();
        for (String s: strs) {
            res.append(s.length()).append("#").append(s);
        }
        return res.toString();
    }

    public List<String> decode(String str) {
        List<String> res = new ArrayList<>();
        int i = 0;
        while (i < str.length()) {
            int j = i;
            while (str.charAt(j) != '#') {
                j++;
            }
            int length = Integer.valueOf(str.substring(i,j));
            i = j + 1 + length;
            res.add(str.substring(j + 1, i));
        }
        return res;
    }
}
